tests/future/batch_semaphore.rs

```diff
@@ -512,52 +512,63 @@ fn bugged_cleanup_would_cause_deadlock() {
     )
 }
 
-// This test exercises the following scenario where SEM is a BatchSemaphore with N permits.
-// 1. Initially the semaphore has 0 permits
-// 2. Task T1 tries to acquire 1 permit, gets added as the first Waiter
-// 3. Task T2 tries to acquire 1 permit, gets added as the second Waiter
-// 4. Task T0 releases 1 permit
-// 5. Task T1 drops its Acquire handle without calling poll()
+// This test exercises scenarios to ensure that the BatchSemaphore behaves correctly in the presence
+// of tasks that drop an `Acquire` guard without waiting for the semaphore to become available.
 //
-// At this point, T2 should be woken up and should get the permit.
-// Unfortunately, in an earlier version of BatchSemaphore, this was not happening because
-// the Drop handler for Acquire was only returning permits to the semaphore, but not
-// waking up the next waiter in the queue.
+// The general idea is that there are 3 types of tasks: `EarlyDrop`, `Hold` and `Release` tasks
+// (determined by the `Behavior` enum).
+// S1. The semaphore initially has 0 permits.
+// S2. The main task spawns a set of tasks, specifying the behavior of each, and the number of permits it should request
+// S3. Each task polls the semaphore once when it is created, in order to get added as a Waiter.
```

Comment:
Nit: "it" here refers to the semaphore, but we want it to refer to the task.

Reply:
Fixed